src/transformers/trainer.py

```diff
@@ -1902,18 +1778,8 @@ def _inner_training_loop(
                     # Optimizer step
                     optimizer_was_run = True
                     if is_torch_tpu_available():
-                        if self.do_grad_scaling:
-                            self.scaler.step(self.optimizer)
-                            self.scaler.update()
-                        else:
-                            # tpu-comment: accelerate wrapped optimizers call xm.optimizer_step
-                            self.optimizer.step()
-                    elif self.do_grad_scaling:
-                        scale_before = self.scaler.get_scale()
-                        self.scaler.step(self.optimizer)
-                        self.scaler.update()
-                        scale_after = self.scaler.get_scale()
-                        optimizer_was_run = scale_before <= scale_after
+                        # tpu-comment: accelerate wrapped optimizers call xm.optimizer_step
+                        self.optimizer.step()
                     else:
                         self.optimizer.step()
                         optimizer_was_run = not self.accelerator.optimizer_step_was_skipped
```

Comment:
Correct me if I'm wrong, but we can now remove this `if` entirely and just have `self.optimizer.step` and the check for if it were skipped?

Reply:
Nice catch!